tests/yam/agama/boot_agama.pm

```diff
@@ -50,4 +50,14 @@ sub run {
     $agama_up_an_running->expect_is_shown();
 }
 
+sub post_fail_hook {
+    my ($self) = shift;
+    select_console('log-console');
+    $self->SUPER::post_fail_hook;
+
+    select_console 'root-console';
+    # "agama logs store" gathers output from dmesg, journalctl and y2logs.
+    save_and_upload_log('agama logs store', "/tmp/agama-logs.tar.gz");
```

Comment:
can we get logs from agama if doesn't even start, I mean, I don't if makes sense to have this here for booting errors.
Another question, if agama logs gather stuff from multiple places why we save the journal as well in agama_base?

Reply:
We can save logs even no failure happened, while we still need to get logs when failure happened, so need the hook. If don't want to save other logs, we can remove the $self->SUPER::post_fail_hook;, and judge or not the point that agama service starts to get the logs.
Save journal log in case developer need that when failure happened, this is not saved in agama_base.

Reply:
Regarding journalctl, then remove comment in line 59, it might confuse someone else.
About if this block is needed, we can save logs for agama in the next test module, the one that does the installation or auto-installation, this is only about booting (doesn't matter if it is on failure or always do it, actually we will need to have them always, but not here, agama might not give you anything if you are in grub phase).

Reply:
In fact, I think it is needed, because this module run the whole process of installation, if failed during installation we can get agama service log to analyze.

Reply:
Removed line 51.

Reply:
no, this module only get to the screen when the installation start, auto or interactive, the responsability to install is delegated to subsequent modules.

Reply:
Yes, it hasn't start to install yet. Ok, I will remove the post_fail_hook since not so much useful for agama service log.